web_extension/options/options_page.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+class OptionsSynchronizer {
+  constructor () {
+    this.options = new Options()
+  }
+
+  get checks () {
+    return document.querySelector('#photographic_options')
+                   .querySelectorAll('input[type=checkbox]')
+  }
+
+  get selects () {
+    return document.querySelector('#photographic_options')
+                   .querySelectorAll('select')
+  }
+
+  async read () {
+    let that = this
+    return this.options.read().then(function(){
+      for (let check_box of that.checks)
+        check_box.checked = that.options[check_box.dataset.namespace][check_box.id]
+
+      for (let select_box of that.selects)
+        for (let option of select_box.options)
+          option.selected = (
+            option.value == that.options[select_box.dataset.namespace][select_box.id]
+          )
+    })
+  }
+
+  async write () {
```

Comment:
no `async`?

Reply:
options.write() is async, 🌡

Reply:
if you aren't `await`ing though you don't need the `async` :D